UnitTests/AppAuthTV/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,279 @@
+    NSArray<NSString *> *components = [paramString componentsSeparatedByString:@"="];
+
+    if (components.count == 2) {
+      bodyParameters[components[0]] = components[1];
+    }
+  }
+
+  return bodyParameters;
+}
+
+/*! @brief Tests the initializer
+ */
+- (void)testInitializer {
+  NSArray<NSString *> *testScopes = @[ kTestScope, kTestScopeA ];
+  NSString *testScopeString = [OIDScopeUtilities scopesWithArray:testScopes];
+  NSDictionary<NSString *, NSString *> *testAdditionalParameters =
+      @{kTestAdditionalParameterKey : kTestAdditionalParameterValue};
+
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:testScopes
+                                          additionalParameters:testAdditionalParameters];
+
+  XCTAssertEqualObjects(authRequest.clientID, kTestClientID);
+  XCTAssertEqualObjects(authRequest.clientSecret, kTestClientSecret);
+  XCTAssertEqualObjects(authRequest.scope, testScopeString);
+  XCTAssertEqualObjects(authRequest.additionalParameters, testAdditionalParameters);
+  XCTAssertEqualObjects(authRequest.responseType, OIDResponseTypeCode);
+  XCTAssertEqualObjects(authRequest.redirectURL, [[NSURL alloc] init]);
+}
+
+/*! @brief Tests the @c NSCopying implementation by round-tripping an instance through the copying
+ * process and checking to make sure the source and destination both contain the
+ * @c TVAuthorizationEndpoint
+ */
+- (void)testCopying {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequest.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+
+  OIDAuthorizationRequest *authRequestCopy = [authRequest copy];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequestCopy.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+}
+
+/*! @brief Tests the @c NSSecureCoding implementation by round-tripping an instance through the
+ * coding process and checking to make sure the source and destination both contain the
+ * @c TVAuthorizationEndpoint
+ */
+- (void)testSecureCoding {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
```

Comment:
Same as above, create local variable for [self testServiceConfiguration].

Reply:
Unneeded now since I am using the endpoint constant string in the assert